theano/tests/unittest_tools.py

```diff
@@ -447,11 +447,16 @@ def attempt_multiple_times(*args, **kwargs):
         return attempt_multiple_times
 
 
-def expectedFailure_fast(f):
+def assertFailure_fast(f):
     """A Decorator to handle the test cases that are failing when
     THEANO_FLAGS =cycle_detection='fast'.
     """
     if theano.config.cycle_detection == 'fast':
-        return unittest.expectedFailure(f)
+        class TestAssertion(unittest.TestCase):
+            def runTest(self, *args, **kwargs):
+                with self.assertRaises(Exception):
+                    f(*args, **kwargs)
+        test_assertion = TestAssertion()
+        return test_assertion
```

Comment:
I think you can simplify this a bit by importing
```
from nose.tools import assert_raises
```
and then
```
def assertFailure_fast(f):
    if theano.config.cycle_detection == 'fast':
        def new_test(*args, **args):
            assert_raises(f, *args, **kwargs)
        return new_test
    else:
        return f
```
rather than defining and instanciating a whole new class